spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java

```diff
@@ -117,32 +119,25 @@ public FlywayConfiguration(FlywayProperties properties,
 			this.flywayCallbacks = flywayCallbacks.getIfAvailable(Collections::emptyList);
 		}
 
-		@PostConstruct
-		public void checkLocationExists() {
-			if (this.properties.isCheckLocation()) {
-				Assert.state(!this.properties.getLocations().isEmpty(),
-						"Migration script locations not configured");
-				boolean exists = hasAtLeastOneLocation();
-				Assert.state(exists,
-						() -> "Cannot find migrations location in: " + this.properties
-								.getLocations()
-						+ " (please add migrations or check your Flyway configuration)");
-			}
-		}
-
-		private boolean hasAtLeastOneLocation() {
-			for (String location : this.properties.getLocations()) {
-				if (this.resourceLoader.getResource(location).exists()) {
-					return true;
-				}
-			}
-			return false;
-		}
-
 		@Bean
 		@ConfigurationProperties(prefix = "spring.flyway")
 		public Flyway flyway() {
-			Flyway flyway = new SpringBootFlyway();
+			Flyway flyway = createFlyway();
+			flyway.setCallbacks(this.flywayCallbacks
+					.toArray(new FlywayCallback[this.flywayCallbacks.size()]));
+			String[] locations = resolveLocations(flyway.getDataSource());
+			checkLocationExists(locations);
+			flyway.setLocations(locations);
+			return flyway;
+		}
+
+		private Flyway createFlyway() {
+			Flyway flyway = new Flyway() {
+				@Override
+				public void setLocations(String... locations) {
```

Comment:
@wilkinsona I'm doing so because `@ ConfigurationProperties ` is overriding the value. May the `SpringBootFlyway` class approach looks better? WDYT?

Reply:
Ah, yes. Of course. I'd forgotten that we still bind directly to the `Flyway` instance. You're right. Let's keep the `SpringBootFlyway` class please.